src/Jobs/UpdateCoupon.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types = 1);
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Jobs;
+
+defined( 'ABSPATH' ) || exit();
+
+/**
+ * Class UpdateCoupon
+ *
+ * Submits WooCommerce coupon to Google Merchant Center and/or updates the existing one.
+ *
+ * Note: The job will not start if it is already running.
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs
+ */
+class UpdateCoupon extends AbstractCouponSyncerJob implements 
+    StartOnHookInterface {
+
+    /**
+     * Get the name of the job.
+     *
+     * @return string
+     */
+    public function get_name(): string {
+        return 'update_coupon';
+    }
+
+    /**
+     * Process an item.
+     *
+     * @param int[] $coupon_ids
+     *
+     * @throws SyncerException If an error occurs. The exception will be logged by ActionScheduler.
+     * @throws JobException If invalid or non-existing products are provided. The exception will be logged by ActionScheduler.
+     */
+    public function process_items( $coupon_ids ) {
+        foreach ( $coupon_ids as $coupon_id ) {
+            $coupon = $this->$wc->may_get_coupon( $coupon_id );
+            if ( 
+                $coupon instanceof WC_Coupon &&
+                $this->coupon_helper->is_sync_ready( $coupon ) ) {
+                $this->$coupon_syncer->update( $coupon );
+            }
```

Comment:
There's no `$wc` or `$coupon_syncer` variables. I think these are supposed to be without the `$` sign.

We also need an import for the `WC_Coupon` class.

```suggestion
            $coupon = $this->wc->may_get_coupon( $coupon_id );
            if ( 
                $coupon instanceof WC_Coupon &&
                $this->coupon_helper->is_sync_ready( $coupon ) ) {
                $this->coupon_syncer->update( $coupon );
            }
```

Reply:
Done.